src/internal_modules/roc_fec/block_writer.cpp

```diff
@@ -156,27 +156,32 @@ status::StatusCode BlockWriter::write(const packet::PacketPtr& pp) {
     return status::StatusOK;
 }
 
-bool BlockWriter::begin_block_(const packet::PacketPtr& pp) {
+status::StatusCode BlockWriter::handle_status_(status::StatusCode code,
+                                               const char* context) {
+    if (code != status::StatusOK) {
+        roc_log(LogError, "fec block writer: %s failed with status code %d", context,
+                code);
+        alive_ = false;
+    }
+    return code;
+}
```

Comment:
I think this helper actually harms readability than improves it.

Yes, it reduces number of lines of code, but extracted code is trivial and it was not really increasing cognitive load. When you read it, you immediately see the idiom (if + logging) and easily skip it.

On the other hand, the helper introduces an indirection level and you need to check out what it's doing. It would be worth it if the help was hiding some complicated logic, but that's not the case.

I usually try to avoid helpers for trivial pieces, less lines does not always mean better code.